### PR TITLE
Convert remaining docstrings to numpydoc style

### PR DESCRIPTION
**Pull request by K-Meech**
## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [X] Other

**Why is this PR needed?**
Some docstrings are still in reST format.

**What does this PR do?**
Updates all remaining docstrings to `numpydoc` format.

## References

For https://github.com/brainglobe/brainglobe-utils/issues/19

## How has this PR been tested?

All tests pass locally.

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

No

## Checklist:

- [X] The code has been tested locally
- [X] Tests have been added to cover all new functionality (unit & integration)
- [X] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: brainglobe_utils/pandas/misc.py [hunk 1]
```diff
@@ -4,18 +4,34 @@
...
+    ----------
+    df : pd.DataFrame
+        Any dataframe
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe with Inf replaced with NaN.
     """
     df = df.replace(np.inf, np.nan)
```
Comment (willGraham01) [inline brainglobe_utils/pandas/misc.py:36]: Not sure why we have a whole function that creates a copy, assigns it to the input, then returns the input, but it was here before so I won't question it.

Though it does seem like overkill when 

```python
df.replace(np.inf, np.nan, inplace=True)
```

does literally the same thing.

File: brainglobe_utils/pandas/misc.py [hunk 2]
```diff
@@ -25,14 +41,24 @@ def safe_pandas_concat(df1: pd.DataFrame, df2: pd.DataFrame) -> pd.DataFrame:
     """
     Concatenate two DataFrames without relying on deprecated functionality
     when one of the DataFrames is empty.
+
     If df1 and df2 are non-empty, return the concatenation.
```
Comment (willGraham01) [inline brainglobe_utils/pandas/misc.py:45]:
```suggestion
    If df1 and df2 are non-empty, return the concatenation. df2 will extend df1 in the resulting DataFrame.
```

Something I should have added for clarity when writing this function :sweat_smile:

**Review by willGraham01 — APPROVED** [review]
No problems with the changes, comments are things that I noticed I should've added to the docstrings when writing them in the first place :sweat_smile:

File: brainglobe_utils/pandas/misc.py (hunk 1 above)
Comment (K-Meech) [reply brainglobe_utils/pandas/misc.py:36]: Yes it does seem like overkill! I guess it is needed elsewhere in BrainGlobe

**Pull request merged**